Assets/Scripts/Bahavior/Dash.cs

```diff
@@ -0,0 +1,67 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Dash : MovementBehavior
+{
+    [Header("Animations")]
+    [SerializeField] GameObject spriteObject;
+    [SerializeField] private Animator anim;
+    [Header("Stats")]
+    [SerializeField] float dashPower;
+    [SerializeField] float dashTime;
+    [SerializeField] float dashCoolDown;
+
+    //Must-have variables for movements
+    [SerializeField] IDashInput dashInput;
+    [SerializeField] TrailRenderer tr;
+
+    [SerializeField] Transform groundCheck;
+    [SerializeField] float checkRadius;
+    [SerializeField] LayerMask whatIsGround;
+
+    private bool isGrounded;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        rb = movingObject.GetComponent<Rigidbody2D>();
+        /*anim = spriteObject.GetComponent<Animator>();*/
+        dashInput = GetComponent<IDashInput>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        Dashing();
+    }
+
+    protected virtual void Dashing()
+    {
+        if (PauseMenu.gameIsPaused == true) return;
```

Comment:
1. Use this statement only for player input
2. Ask UIManager to know if the player is using UI or not for this statement

Reply:
Ố kề